### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by kkmsft**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

 /kind cleanup

> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
- Ignore vscode setting.json (.gitignore)
- Fixes to autogeneration to reflect the type.go source file move.
- Lint fixes.
- Go sum - add generated file 

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: client/api/filesystem/v1alpha1/api.pb.go [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
-// source: github.com/kubernetes-csi/csi-proxy/client/api/filesystem/v1alpha1/api.proto
```
Comment (ddebroy) [inline client/api/filesystem/v1alpha1/api.pb.go]: Is this dependent on where protoc-gen-go is executed from?

Comment (ddebroy) [reply client/api/filesystem/v1alpha1/api.pb.go]: some external examples: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/storage/v1/generated.pb.go#L18
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/networking/v1/generated.pb.go#L18

File: .gitignore [hunk 2]
```diff
@@ -1 +1,2 @@
 /build/
+settings.json
```
Comment (wk8) [inline .gitignore]: Not sure I understand what that file is?

Comment (kkmsft) [reply .gitignore]: This is vscode settings file. The goal is to ignore those files for vscode users.

File: internal/server/filesystem/server.go [hunk 3]
```diff
@@ -76,53 +73,38 @@ func (s *Server) validatePathWindows(pathCtx internal.PathContext, path string)
 	} else if pathCtx == internal.POD {
 		prefix = s.kubeletPodPath
 	} else {
-		return fmt.Errorf("Invalid PathContext: %v", pathCtx)
+		return fmt.Errorf("invalid PathContext: %v", pathCtx)
```
Comment (jingxu97) [inline internal/server/filesystem/server.go]: maybe add details about the error, like PathContext must be either PLUGIN or POD ?

File: internal/server/filesystem/server.go [hunk 4]
```diff
@@ -76,53 +73,38 @@ func (s *Server) validatePathWindows(pathCtx internal.PathContext, path string)
...
 
 	if !strings.HasPrefix(path, prefix) {
-		return fmt.Errorf("Path: %s is not within context path: %s", path, prefix)
+		return fmt.Errorf("path: %s is not within context path: %s", path, prefix)
 	}
 
 	return nil
 }
 
-func (s *Server) abs(pathCtx internal.PathContext, path string) (string, error) {
```
Comment (jingxu97) [inline internal/server/filesystem/server.go]: what is the original purpose of this function?

**Pull request merged**